lib/host/ssvm_process/processfunc.cpp

```diff
@@ -85,6 +86,21 @@ SSVMProcessRun::body(Runtime::Instance::MemoryInstance *MemInst) {
   Env.StdErr.clear();
   Env.ExitCode = static_cast<uint32_t>(-1);
 
+  /// Check white list of commands.
+  if (!Env.AllowedAll &&
+      Env.AllowedCmd.find(Env.Name) == Env.AllowedCmd.end()) {
+    Env.Name.clear();
+    Env.Args.clear();
+    Env.Envs.clear();
+    Env.StdIn.clear();
+    std::string Msg = "Permission denied.\n";
```

Comment:
You should provide more details error message such as `Permission denied: Command "OOO" is not in the white list. Please use --allow-command=OOO or --allow-command-all to add "OOO" command in the white list.`